go/vt/vtadmin/vtsql/config.go

```diff
@@ -97,33 +97,48 @@ func (c *Config) Parse(args []string) error {
 			"a Username and Password. Templates are given the context of the vtsql.Config, and primarily "+
 			"interoplate the cluster name and ID variables.")
 	effectiveUser := fs.String("effective-user", "", "username to send queries on behalf of")
-
+	credentialsUsername := fs.String("credentials-username", "",
+		"A string specifying the Username to use for authenticating with vtgate. "+
+			"Used with credentials-password in place of credentials-path-tmpl, in cases where providing a static file cannot be done.")
+	credentialsPassword := fs.String("credentials-password", "",
+		"A string specifying a Password to use for authenticating with vtgate. "+
+			"Used with credentials-username in place of credentials-path-tmpl, in cases where providing a static file cannot be done.")
 	if err := fs.Parse(args); err != nil {
 		return err
 	}
 
-	var creds *grpcclient.StaticAuthClientCreds
+	// First set credentials to values potentially supplied by credentials-password and credentials-username
+	c.Credentials = &StaticAuthCredentials{
```

Comment:
Two things:

1. This means that now `c.Credentials` will _never_ be `nil`, where previously it would only be non-nil if `credentials-path-tmpl` was specified. Is that okay?
2. The way I read the flag help ("Used with credentials-password in place of credentials-path-tmpl") made me think that if both were set, user/pass would override anything in tmpl, but the code does the reverse. Should (a) we keep the code that way or (b) change the wording to indicate the reverse?

Reply:
Those are both great point and this PR should be updated to reflect them! 😄 Updating!